protob/messages.proto

```diff
@@ -1437,3 +1454,125 @@ message DebugLinkMemoryWrite {
 message DebugLinkFlashErase {
 	optional uint32 sector = 1;
 }
+
+/**
+ * Response: Contains a Cardano addresses derived from device private seed
+ * @prev CardanoGetAddresses
+ */
+message CardanoAddresses {
+	repeated CardanoAddress addresses = 1;
+}
+
+/**
+ * Request: Ask device for Cardano addresses
+ * @next CardanoAddresses
+ * @next Failure
+ */
+ message CardanoGetAddresses {
+	repeated CardanoDerivationPath paths = 1;		// BIP-32-style paths to derive the key from master node
+	optional bool show_display = 2;				// optionally prompt for confirmation on trezor display
+}
+
+/**
+ * Request: Ask device for Cardano address
+ * @next CardanoAddress
+ * @next Failure
+ */
+ message CardanoGetAddress {
+	required CardanoDerivationPath path = 1;		// BIP-32-style path to derive the key from master node
+	optional bool show_display = 2;				// optionally prompt for confirmation on trezor display
+}
+
+/**
+ * Request: Ask device to sign Cardano message
+ * @next CardanoMessageSignature
+ * @next Failure
+ */
+message CardanoSignMessage {
+	repeated uint32 address_n = 1;				// BIP-32 path to derive the key from master node
+	required bytes message = 2;				// message to be signed
+}
+
+/**
+ * Request: Ask device for public key corresponding to address_n path
+ * @next CardanoXpubKey
+ * @next Failure
+ */
+ message CardanoGetXpubKey {
+	repeated uint32 address_n = 1;				// BIP-32 path to derive the key from master node
+}
+
+/**
+ * Response: Contains public key derived from device private seed
+ * @prev CardanoGetXpubKey
+ */
+ message CardanoXpubKey {
+	required string xpub_key = 1;				// Xpub key
+	required string public_key = 2;				// Public key
+	required string chain_code = 3;				// Chain code
+	repeated uint32 address_n = 4;				// Path
+	required string root_hd_passphrase = 5;			// Hd passphrase for root in hex format
+}
+
+/**
+ * Request: Ask device to verify Cardano message
+ * @next Success
+ * @next Failure
+ */
+message CardanoVerifyMessage {
+	optional bytes public_key = 1;				// Public key which was used to sign message
+	optional bytes signature = 2;				// signature to verify
+	optional bytes message = 3;				// message to verify
+}
+
+/**
+ * Response: Signed Cardano message
+ * @prev CardanoSignMessage
+ */
+message CardanoMessageSignature {
+	optional string address = 1; 				// address used to sign the message
+	optional bytes signature = 2;				// signature of the message
+}
+
+/**
+ * Request: Ask device to sign Cardano transaction
+ * @next CardanoSignedTransaction
+ * @next CardanoTxRequest
+ * @next Failure
+ */
+message CardanoSignTransaction {
+	repeated CardanoTxInputType inputs = 1;  		// inputs to be used in transaction
+	repeated CardanoTxOutputType outputs = 2;		// outputs to be used in transaction
+	repeated bytes attributes = 3; 				// transaction attributes
```

Comment:
What is the purpose of `attributes`? Can we use protobuf enum/submessage instead?

Reply:
Currently it is not defined and attributes must be empty. But cardano team is mentioning that in version 1 (current is 0), attributes will be used for additional data storage. More about it can be found [here](https://github.com/input-output-hk/cardano-sl/blob/8d25c2ad3ca2354af8f8c43a2972d1b9a31bf440/docs/tx/processing.md)

Reply:
Let's drop this attribute then. We can always add it later (preferably in form of a submessage, not arbitrary bytes value).